deeplake/auto/structured/dataframe.py

```diff
@@ -20,6 +36,130 @@ def __init__(self, source):
         if not isinstance(self.source, pd.DataFrame):
             raise Exception("Source is not a pandas dataframe object.")
 
+        self.creds = creds
+        self.creds_key = creds_key
+        self._initialize_params(column_params)
+
+    # def _sanitize_tensor(self, input: str):
+    #     """Sanitize a string to be a valid tensor name
+
+    #     Args:
+    #         input (str): A string that will be sanitized
+
+    #     Returns:
+    #         str: A string with the sanitized tensor name
+    #     """
+
+    #     invalid_chars = ["[", "]", "@", ".", ",", "?", "!", "/", "\\", "#", "'", '"']
+    #     new_char = "_"
+    #     for char in invalid_chars:
+    #         input = input.replace(char, new_char)
+
+    #     return input
+
+    def _initialize_params(self, column_params):
+        column_params = column_params if column_params is not None else {}
+
+        keys = list(self.source.columns)
+
+        for key in keys:
+            if key in column_params.keys():
+                column_params[key]["name"] = column_params[key].get(
+                    "name", sanitize_tensor_name(key)
+                )  # Get the specified name, and if it doesn't exist, use the sanitized column name
+            else:
+                column_params[key] = {
+                    "name": sanitize_tensor_name(key)
+                }  # If column parameters are not specified for a column, use the sanitized column name
+
+        self.column_params = column_params
```

Comment:
Consider following refactor, call sanitize method only if necessary:

```suggestion
        column_params = column_params or {}
        for key in self.source.columns:
            params = column_params.get(key)
            if params:
                if "name" not in params:
                    params["name"] = sanitize_tensor_name(key)
            else:
                 column_params[key] = {"name": sanitize_tensor_name(key)}
        self.column_params = column_params
```

Reply:
@farizrahman4u Doesn't `params` need to be assigned to `column_params` somewhere.

Are you not doing that because line 61 is passed by reference `params = column_params.get(key)`???

Reply:
Yes. params is a ref, changes to it will be reflected in column_params